_posts/2024-09-05-releasing-apps.md

```diff
@@ -0,0 +1,85 @@
+
+Apps are constrained by the stores, not shipping our releases to customers before taking their time reviewing. It takes a good amount longer for adoption to then ramp up, meaning it takes a long time to create a fix and address it. Even if we had incredible test coverage, it’s near-impossible to test every given scenario your app will be run under in production.
+
+A branching structure growing in popularity for mobile apps nowadays is a combination of trunk-based development and release branches. Let’s explain:
+
+Developers do their work in small, manageable chunks, and merge into a trunk branch: `main`. Every week (or fortnight in our case), a workflow kicks off to cut a _release branch_. This release branch behaves similarly to the GitFlow release branch—it’s fixing our base at a given point, allowing us to stabilize before releasing. Fixes for the release could go one of two ways - based on release and merged into the release branch, or based on `main`, and back-ported to the release branch.
+
+Merging a release branch back in to where it came from was one of the time sinks we were looking to get rid of, removing the need to resolve merge conflicts. Instead, we aimed for fixing on `main`, and _never_ merging the release branch back in.
+
+![image](/assets/img/2024-09-05-trunk-ish-branching.png)
+*Trunk branching with release branches*
+
+To make this “back-porting” simple for developers, we introduced an automatic cherry-picking workflow. When bugs come up in a release before deployment, pull requests with fixes would land in `main`, first. These pull requests have a special label which another piece of automation picks up on, knowing to cherry-pick the newly merged work on to the open release branch!
+
+This works the majority of the time, but what about merge conflicts? In our old school GitFlow model, resolving conflicts ended up being the responsibility of the release champion, who may or may not have all the context needed. With our cherry-picking flow, the conflicts end up having to be resolved more with more granularity the _other way around_.
+
+Developers fix their bugs on `main`, meaning the code they need to modify to address the issue has moved on from the base the release branch was cut from. If a fix can’t be cleanly cherry-picked onto the release branch, the author of the fix is prompted to create a pull request _per-fix_, addressing more granular conflicts as they come up!
+
+### Building a release process for portability
+Across M&S we have more than 100 mobile applications in their own individual repositories. A lot of these have their own, manual release process, and a consistent branching structure doesn’t exist. We knew this coming in to the reimagining of our releasing process, and built it from the ground up with _portability_ in mind.
```

Comment:
This paragraph feels really, really important. We've buried the lede (wow we can solve the branching and release process for over 100 repositories!). I think re-wording this paragraph to really land the point could be worth it (I've tried to use a what, so what, now what approach. 

```suggestion
Across M&S we have more than 100 mobile applications in their own individual repositories. Each has inconsistent branching and different manual release processes. All contributing to slower releases impacting our customers and colleagues. Built from the ground with _conventions_ and _portability_ in mind, our new process can solve branching and releasing across many code bases and teams, once and for all. Here's how we did it:
```

The rewording isn't perfect and you have your own voice. But I think it is worth nailing the impact of the inconsistencies and how your work has made improvements

Reply:
Great shout. Rejigged this a bit in [57b4953](https://github.com/marksandspencer/marksandspencer.github.io/pull/18/commits/57b4953e3f1847d277d9ba77f8f939ef502c27be), where I've applied your suggestion and tweaked it a bit to be impactful.